controllers/nnf_workflow_controller_container_helpers.go

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright 2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	dwsv1alpha2 "github.com/HewlettPackard/dws/api/v1alpha2"
+	nnfv1alpha1 "github.com/NearNodeFlash/nnf-sos/api/v1alpha1"
+	mpicommonv1 "github.com/kubeflow/common/pkg/apis/common/v1"
+	mpiv2beta1 "github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v2beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type nnfUserContainer struct {
+	workflow *dwsv1alpha2.Workflow
+	profile  *nnfv1alpha1.NnfContainerProfile
+	nnfNodes []string
+	volumes  []nnfContainerVolume
+	username string
+	uid, gid int64
+	r        *NnfWorkflowReconciler
```

Comment:
Send in the parts of 'r' that you need.  So use these two fields, instead:
```
 clnt client.Client
log logr.Logger
```

Then set those fields with r.Client and r.Log.

Reply:
Perfect. Thanks!